src/Hedgehog/Gen.fs

```diff
@@ -315,28 +300,30 @@ module Gen =
     let private atLeast (n : int) (xs : List<'a>) : bool =
         n = 0 || not (List.isEmpty (List.skip (n - 1) xs))
 
-    /// Generates a list between 'n' and 'm' in length.
-    let list' (n : int) (m : int) (g : Gen<'a>) : Gen<List<'a>> =
-        ofRandom <| random {
-            let! k = Random.range n m
-            let! xs = Random.replicate k (toRandom g)
-            return Shrink.sequenceList xs
-                |> Tree.filter (atLeast (min n m))
-        }
+    /// Generates a list using a 'Range' to determine the length.
+    let list' (range : Range<int>) (g : Gen<'a>) : Gen<List<'a>> =
+        ofRandom
+        <| (Random.sized
+        <| fun size -> random {
+               let! k = Random.integral range
+               let! xs = Random.replicate k (toRandom g)
+               return Shrink.sequenceList xs
+                   |> Tree.filter (atLeast (Range.lowerBound size range))
+           })
 
     /// Generates a list of random length. The maximum length depends on the
     /// size parameter.
     let list (g : Gen<'a>) : Gen<List<'a>> =
-        sized (fun size -> list' 0 size g)
+        sized (fun size -> list' (Range.constant 0 size) g)
 
     /// Generates a non-empty list of random length. The maximum length depends
     /// on the size parameter.
     let list1 (g : Gen<'a>) : Gen<List<'a>> =
-        sized (fun size -> list' 1 size g)
+        sized (fun size -> list' (Range.constant 1 size) g)
```

Comment:
I think we should remove `list` and `list1` and rename `list'` to `list`, as the range combinators give a much richer way to generate all kinds of lists.